[improve][doc] Deduplication can have effects in performance and is not activated at client level (#15368)
apache/pulsar: site2/docs/cookbooks-deduplication.md

sidebar_label: Message deduplication 
---

When **Message deduplication** is enabled, it ensures that each message produced on Pulsar topics is persisted to disk *only once*, even if the message is produced more than once. Message deduplication is handled automatically on the broker side. 

> Message deduplication could affect performance in the brokers during informational snapshot.

Reviewer: ```suggestion
Message deduplication could affect the performance of the brokers during informational snapshots.
```

> To use message deduplication in Pulsar, you need to configure your Pulsar brokers, namespaces or topics and it is recommended to modify configuration in the clients, setting send timeout to infinity.

Reviewer: ```suggestion
To use message deduplication in Pulsar, you need to configure your Pulsar brokers, namespaces, or topics. It is recommended to modify the configuration in the clients, for example, setting send timeout to infinity.
```

## How it works

> You can enable or disable message deduplication at broker, namespace or topic level. By default, it is disabled on all brokers, namespaces or topics. You can enable it in the following ways:

Reviewer: ```suggestion
You can enable or disable message deduplication at broker, namespace, or topic level. By default, it is disabled on all brokers, namespaces, or topics. You can enable it in the following ways:
```

## Pulsar clients

> If you enable message deduplication in Pulsar brokers, namespaces or topics, it is recommended to make the client retry infinitely the messages until it succeed, otherwise is possible to break the ordering guarantee as some request can time out and the application wont know if the request was successfully added to the topic or not. 

Reviewer: ```suggestion
If you enable message deduplication in Pulsar brokers, namespaces, or topics, it is recommended to make the client retry infinitely the messages until it succeeds, otherwise it is possible to break the ordering guarantee as some requests may time out and the application does not know whether the request is successfully added to the topic or not. 
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true.
https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

Author: Thanks for the link. I will review it next.

<!--DOCUSAURUS_CODE_TABS-->
<!--Java clients-->

> Not to break the guarantee order on a [Java producer](client-libraries-java.md#producers) sending to a topic with message deduplication active, set the producer name using the `producerName` setter, and set the timeout to `0` using the `sendTimeout` setter. 

Reviewer: what is the meaning of "Not to break the guarantee order on a Java producer sending to a topic with message deduplication active"? 

what are the logical relationships between the 3 parts of this whole sentence?

would u mind re-writing this sentence? Thanks

Reviewer 2: ```suggestion
To ensure the guarantee order on a [Java producer](client-libraries-java.md#producers) sending to a topic with message deduplication enabled, set the producer name using the `producerName` setter, and set the timeout to `0` using the `sendTimeout` setter. 
```

@AlvaroStream did you want to say something like this?
